frontend/src/metabase/query_builder/components/DataSelector/DataSelectorTablePicker/DataSelectorTablePicker.tsx

```diff
@@ -180,11 +180,16 @@ const Header = ({
   <HeaderContainer>
     <HeaderClickable onClick={onBack}>
       {onBack && <Icon name="chevronleft" size={18} />}
-      <HeaderDatabaseName>{selectedDatabase.name}</HeaderDatabaseName>
+      <HeaderDatabaseName data-testid="source-database">
+        {selectedDatabase.name}
+      </HeaderDatabaseName>
     </HeaderClickable>
 
     {selectedSchema?.name && schemas.length > 1 && (
-      <HeaderSchemaName>- {selectedSchema.displayName()}</HeaderSchemaName>
+      <>
+        <HeaderSchemaDivider>-</HeaderSchemaDivider>
+        <span data-testid="source-schema">{selectedSchema.displayName()}</span>
+      </>
```

Comment:
I could've let this one go, but it really bothered me that we're faking the distance on the left side of the divider `-` with the margin, while using a regular space on the right.

Also saw a test that asserted the following string was in the DOM "- schemaName" which felt wrong.
This is not the ideal solution either and this whole component could be improved, but it does the job for now, IMO.